src/stats.rs

```diff
@@ -66,3 +66,40 @@ pub fn calc_stats(results: Vec<Result<u128, String>>) -> Stats {
         success_count: idx,
     }
 }
+
+#[cfg(test)]
+mod tests {
+    #[allow(unused_imports)]
+    use super::*;
+
+    #[test]
+    fn median_sort() {
+        let v = vec![2, 1, 4, 3];
+        assert_eq!(median(v), 3);
+    }
+
+    #[test]
+    fn median_odd() {
+        let v = vec![0, 4, 123, 5, 8];
+        assert_eq!(median(v), 5);
+    }
+
+    #[test]
+    fn calc() {
+        let mut results = Vec::<Result<u128, String>>::new();
+        let r1 = Result::Ok(3);
+        let r2 = Result::Err("Error".to_owned());
+        let r3 = Result::Ok(5);
+        let r4 = Result::Ok(7);
+        results.push(r1);
+        results.push(r2);
+        results.push(r3);
+        results.push(r4);
```

Comment:
Seems like a pretty good use case for `vec!`

Reply:
good catch, I updated it